middleware/gateway/source/group/outbound/handle.cpp

```diff
@@ -114,10 +114,12 @@ namespace casual
                                  state.multiplex.send( destination.ipc, reply);
                               });
 
-                              tcp::send( state, descriptor, message);
+                              // if we fail to send we want to invoke the error callback immediately
+                              if( ! tcp::send( state, descriptor, message).valid())
+                                 state.route.consume( message.correlation).error();
```

Comment:
Should this be done generically by `tcp::send`. Naively, this could happen in other "flows"? Whats your thoughts?